dataset/test/dataset_test.cpp

```diff
@@ -154,6 +156,18 @@ TEST(DatasetTest, setMask) {
   ASSERT_EQ(d.masks().size(), 2);
 }
 
+TEST(DatasetTest, setData_with_mismatched_dims) {
```

Comment:
Is there really no existing test for this?

Reply:
Apparently not, I searched through all tests expecting a `DimensionError", the closest test was `DataArray_unaligned`.

Reply:
What about https://github.com/scipp/scipp/blob/f00d9351ffc7425bd31cc8ed971f896c6ab94e2e/dataset/test/dataset_test.cpp#L212-L223

Reply:
Ah, the `DataArrayView` threw me as I assumed it was testing a separate detail to `DataSet`

If I drop the new test, would it be worth renaming this unit to `setData` to match the test above (`setData_keeps_attributes`) too?

Reply:
It may actually make sense to keep your new test as well, since it is doing something slightly different. Just place them adjacent?